feat: add payment client and sp functions

owen-reorg opened the pull request:
### Description

Add payment client and sp functions.

### Rationale

Add payment client in new SDK, make it easier to invoke payment related queries and transactions.
Add more functions in Storage API.

### Example

```golang
// create payment account
paymentAccountsBeforeCreate, err := cli.GetPaymentAccountsByOwner(ctx, account.GetAddress().String())
s.Require().NoError(err)
txHash, err := cli.CreatePaymentAccount(ctx, account.GetAddress().String(), types2.TxOption{})
s.Require().NoError(err)
t.Logf("Acc: %s", txHash)
waitForTx, err := cli.WaitForTx(ctx, txHash)
s.Require().NoError(err)
t.Logf("Wair for tx: %s", waitForTx.String())

paymentAccountsByOwnerAfterCreate, err := cli.GetPaymentAccountsByOwner(ctx, account.GetAddress().String())
s.Require().NoError(err)
s.Require().Equal(len(paymentAccountsByOwnerAfterCreate)-len(paymentAccountsBeforeCreate), 1)

// deposit
paymentAddr := paymentAccountsByOwnerAfterCreate[len(paymentAccountsByOwnerAfterCreate)-1].Addr
depositAmount := math.NewIntFromUint64(100)
depositTxHash, err := cli.Deposit(ctx, paymentAddr, depositAmount, types2.TxOption{})
s.Require().NoError(err)
t.Logf("deposit tx: %s", depositTxHash)
waitForTx, err = cli.WaitForTx(ctx, depositTxHash)
s.Require().NoError(err)
t.Logf("Wair for tx: %s", waitForTx.String())

// get stream record
streamRecord, err := cli.GetStreamRecord(ctx, paymentAddr)
s.Require().NoError(err)
s.Require().Equal(streamRecord.StaticBalance.String(), depositAmount.String())

// withdraw
withdrawAmount := math.NewIntFromUint64(50)
withdrawTxHash, err := cli.Withdraw(ctx, paymentAddr, withdrawAmount, types2.TxOption{})
s.Require().NoError(err)
t.Logf("withdraw tx: %s", withdrawTxHash)
waitForTx, err = cli.WaitForTx(ctx, withdrawTxHash)
s.Require().NoError(err)
t.Logf("Wair for tx: %s", waitForTx.String())
streamRecordAfterWithdraw, err := cli.GetStreamRecord(ctx, paymentAddr)
s.Require().NoError(err)
s.Require().Equal(streamRecordAfterWithdraw.StaticBalance.String(), depositAmount.Sub(withdrawAmount).String())

// disable refund
paymentAccountBeforeDisableRefund, err := cli.GetPaymentAccount(ctx, paymentAddr)
s.Require().NoError(err)
s.Require().True(paymentAccountBeforeDisableRefund.Refundable)
disableRefundTxHash, err := cli.DisableRefund(ctx, paymentAddr, types2.TxOption{})
s.Require().NoError(err)
t.Logf("disable refund tx: %s", disableRefundTxHash)
waitForTx, err = cli.WaitForTx(ctx, disableRefundTxHash)
s.Require().NoError(err)
t.Logf("Wair for tx: %s", waitForTx.String())
paymentAccountAfterDisableRefund, err := cli.GetPaymentAccount(ctx, paymentAddr)
s.Require().NoError(err)
s.Require().False(paymentAccountAfterDisableRefund.Refundable)
```

### Changes

Notable changes:
* add payment client

Fynnss commented on client/api_payment.go:
```diff
@@ -0,0 +1,84 @@
+
+	Deposit(ctx context.Context, toAddress string, amount math.Int, txOption *gnfdSdkTypes.TxOption) (string, error)
+	Withdraw(ctx context.Context, fromAddress string, amount math.Int, txOption *gnfdSdkTypes.TxOption) (string, error)
+	DisableRefund(ctx context.Context, paymentAddress string, txOption *gnfdSdkTypes.TxOption) (string, error)
```
As I said here. https://github.com/bnb-chain/greenfield-go-sdk/pull/48#discussion_r1170872321

Suggest not use pointer for gnfdSdkTypes.TxOption for unified interface style.

owen-reorg replied:
fixed

Fynnss commented:
Did you forget to impl `UpdateStorageProviderPrice` interface?

owen-reorg commented:
> Did you forget to impl `UpdateStorageProviderPrice` interface?

added

> The example in the PR description is out of date because we do not use GRPC for Greenfield Client. Pls Update.

updated

Fynnss reviewed (approved):
LGTM
